### PR TITLE
SIMPLY-3206 LCP audiobooks support

### PR DESCRIPTION
**Pull request by vladimirfedorov**
Adds all the necessary objects to play LCP DRM-protected audiobooks.

Actual decryption and license verification is performed on SimplyE side, also LCP audiobooks contain all audio files inside.

### CODE REVIEW

File: NYPLAudiobookToolkit/Core/LCPAudiobook.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            if let spineElement = LCPSpineElement(JSON: resource, index: UInt(index), audiobookID: uniqueIdentifier) {
+                spineElements.append(spineElement)
+            }
+        }
+        spineElements.sort { (a, b) -> Bool in
+            a.chapterNumber < b.chapterNumber
+        }
+        self.spine = spineElements
+        guard let cursor = Cursor(data: spine) else {
+            ATLog(.error, "Cursor could not be cast to Cursor<LCPSpineElement>")
```
Comment (ettore) [inline NYPLAudiobookToolkit/Core/LCPAudiobook.swift:72]: can you add some more context to help debugging, for example name of book or anything that can help identify the book?

File: NYPLAudiobookToolkit/Core/LCPSpineElement.swift [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    private static func elementName(JSON: Any?) -> String? {
+        if let stringName = JSON as? String {
+            return stringName
+        } else if let localizedNames = JSON as? [[String: String]] {
+            return localizedName(localizedNames: localizedNames)
+        }
+        return nil
+    }
+    
+    /// Localized name from the array of localizede names based on current application language code.
```
Comment (ettore) [inline NYPLAudiobookToolkit/Core/LCPSpineElement.swift:100]: typo: `localizede` :)

**Review by ettore — APPROVED** [review]
looks straightforward -- question just to make sure about the deprecated notices you added: are those actually triggered, i.e. is there any usage of those deprecated methods? If so, that would be annoying and I'd rather convert the notices into TODOs with a ticket.

**Review by ErnestFan — APPROVED** [review]

**Comment by vladimirfedorov** [comment]
> usage of those deprecated methods
They are annoying for purpose: they are required and I can't remove them, but they shouldn't be called, Xcode marks them as warnings and attracts attention.

File: NYPLAudiobookToolkit/Core/LCPSpineElement.swift (hunk 2 above)
Comment (vladimirfedorov) [reply NYPLAudiobookToolkit/Core/LCPSpineElement.swift]: fixed... my butterfly keyboard is doing this sometimes

File: NYPLAudiobookToolkit/Core/LCPAudiobook.swift (hunk 1 above)
Comment (vladimirfedorov) [reply NYPLAudiobookToolkit/Core/LCPAudiobook.swift:72]: I can dump JSON data, like at the top of init:

```
ATLog(.error, "LCPAudiobook failed to init from JSON: \n\(JSON ?? "nil")")
```

Comment (vladimirfedorov) [reply NYPLAudiobookToolkit/Core/LCPAudiobook.swift:72]: Not sure if that will be readable, it comes in `message` parameter to NYPLErrorLogger

**Pull request merged**